src/main/java/dev/cdevents/CDEvents.java

```diff
@@ -0,0 +1,100 @@
+package dev.cdevents;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SpecVersion;
+import com.networknt.schema.ValidationMessage;
+import dev.cdevents.config.CustomObjectMapper;
+import dev.cdevents.constants.CDEventConstants;
+import dev.cdevents.events.PipelineRunFinishedCDEvent;
+import dev.cdevents.exception.CDEventsException;
+import dev.cdevents.models.CDEvent;
+import io.cloudevents.CloudEvent;
+import io.cloudevents.core.v03.CloudEventBuilder;
+
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.time.OffsetDateTime;
+import java.util.Set;
+import java.util.UUID;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class CDEvents {
+
+    private CDEvents() {
+    }
+
+    private static ObjectMapper objectMapper = new CustomObjectMapper().customConfiguration();
+    private static Logger log = LoggerFactory.getLogger(CDEvents.class);
+
+
+    /**
+     * @param cdEvent
+     * @return json string of a cdEvent
+     */
+    public static String cdEventAsJson(CDEvent cdEvent) {
+        String asJson = "";
+        try {
+            asJson = objectMapper.writeValueAsString(cdEvent);
+        } catch (JsonProcessingException e) {
+            log.error("Error while mapping cdEvent as Json {}", e.getMessage());
+        }
+        return asJson;
+    }
+
+
+    /**
+     * Creates a CloudEvent from the cdEvent.
+     * @param cdEvent
+     * @return CloudEvent
+     */
+    public static CloudEvent cdEventAsCloudEvent(CDEvent cdEvent) {
+
+        String cdEventJson = cdEventAsJson(cdEvent);
+        if (cdEventJson.isEmpty()) {
```

Comment:
This is certainly a good check to have, but it looks like you're relying on the empty json to catch an error in `cdEventsAsJson` - I think it would be tidier to throw the CDEventsException in `cdEventsAsJson` directly

Reply:
yes throwing CDEventsException from `cdEventsAsJson` itself